### ISSUE TITLE
How to indicate that a message is a translation of another message

### ISSUE THREAD

**Issue by Changaco** [w3c/activitystreams#463]
Please Indicate One:
 * [ ] Editorial
 * [x] Question
 * [ ] Feedback
 * [ ] Blocking Issue
 * [ ] Non-Blocking Issue

Please Describe the Issue:

Is there a standard way to express that an object is a translation of another object? The Activity Vocabulary specification defines an [`inReplyTo` property](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-inreplyto) for replies, but no `isTranslationOf` for translations.

**Comment by evanp** [comment]
Are you looking for natural language values?

https://www.w3.org/TR/activitystreams-core/#naturalLanguageValues

You can use, for example, a `contentMap` to show translations of the same message in different languages.

**Comment by Changaco** [comment]
No, maps are for including multiple languages in a single object, right? What I'm looking for is a way to link two distinct objects together, with a property indicating that the second object is a translation of the first object.

For example [this toot](https://mastodon.xyz/@Liberapay/99751255368634883) is a French translation of [that one](https://mastodon.xyz/@Liberapay/99751124600550302), and we want to encode this fact into the toot's metadata so that we can avoid showing "duplicate" messages to people.

**Comment by nightpool** [comment]
are there any social networks that currently implement this feature? it
seems very speculative.

**Comment by Changaco** [comment]
I don't know of any social network that fully supports multilingualism, and I've never seen one that allows adding a translation after a message has been published.

**Comment by evanp** [comment]
I think FB will let users improve a machine translation, but I'm not sure if they surface that translation to other users.

Maybe look at rdf or dc properties?

**Comment by nclm** [comment]
Facebook does have a multilingual post system, you can define the language of a post when writing and add translations to it. You author them at the same time though, and the result is an unique multilingual post rather than post linked together.

![screenshot](https://user-images.githubusercontent.com/923008/38139361-64c42064-342f-11e8-916f-d8e74917fb54.png)

(sorry for the white overlay one the screenshot)

**Comment by Changaco** [comment]
Since there doesn't appear to be a standard way to express that a message is a translation of another one, should I open a pull request to discuss the possible future standardization of an `isTranslationOf` property?

**Comment by nightpool** [comment]
I can't speak for ActivityStreams, but I think that if you want to model this, the best way to do so would be an Edit to the exiting Note that adds a translation using the contentMap property. I don't think making them separate posts is appropriate. Then the creator of the Note can either endorse or reject the Edit as they wish, via whatever semantics the data interchange layer uses to authenticate these sort of things.

**Comment by Changaco** [comment]
That's a good suggestion. However messages that are translations of other messages already exist and I think it would be good to have a property to describe that reality. Also, introducing `isTranslationOf` wouldn't break anything, any software which doesn't know about it would simply continue to show all messages.

**Comment by Changaco** [comment]
Adding translations to an already-published message would also be problematic in that the message's date would no longer be accurate.

**Comment by evanp** [comment]
I think this is an interesting relationship. If the translation and the original are "the same object", we use the language map mechanism. If they are "not the same object", for example, they have different authors, or some other major difference, having a looser relationship makes sense. This seems like a good topic for a FEP. I'm going to mark this is as needs-fep, and waiting for commenter.

**Comment by bumblefudge** [comment]
I would also note that in the community extensions to the ActivityPub extensions to ActivityStreams, there is a fep called FEP-bad1 which includes previous versions of an edited object-- not sure this is the best way to model it, but if Object 2 is a translation of Object 1, there might be a way to point back to the original (with its distinct timestamp) from the translation as an entry in its object history, the way an object can point to its prior `id` /home after an export/import-to-new-server?  Possibly a red herring but thought I'd mention it to any future FEP authors who read this thread.

**Comment by tesaguri** [comment]
I think you can use the [`hreflang`] property in [FEP-fffd] (<cite>Proxy Objects</cite>) instead of inventing a new property.


[`hreflang`]: <https://www.w3.org/TR/2017/REC-activitystreams-vocabulary-20170523/#dfn-hreflang>
[FEP-fffd]: <https://codeberg.org/fediverse/fep/src/branch/main/fep/fffd/fep-fffd.md>

**Comment by evanp** [comment]
We reviewed this and it seems like this issue still needs a fep.
